File: GanAndroidApp/app/src/main/java/com/xattacker/gan/service/SystemService.java
package com.xattacker.gan.service;

import java.util.Date;

import com.xattacker.binary.TypeConverter;
import com.xattacker.gan.GanAgent;
import com.xattacker.gan.data.FunctionType;
import com.xattacker.gan.data.ResponsePack;

public abstract class SystemService extends ServiceFoundation
{
	protected SystemService(GanAgent aAgent)
	{
		super(aAgent);
	}

	public String getIPAddress()
	{
		String ip = null;
		
		try
		{
			ResponsePack response = send(FunctionType.GET_IP, null);
			if (response != null)
			{
				if (response.getResult())
				{
					ip = new String(response.getContent());
				}
				
				response = null;
			}
		}
		catch (Exception ex)
		{
			ip = null;
		}
		finally
		{
		}
		
		return ip;
	}
	
	public Date getSystemTime()
	{
		Date time = null;
		
		try
		{
			ResponsePack response = send(FunctionType.GET_SYSTEM_TIME, null);
			if (response != null)
			{
				if (response.getResult())
				{
					long timestamp = TypeConverter.byteToLong(response.getContent());
					time = new Date(timestamp);
				}
				
				response = null;
			}
		}
		catch (Exception ex)
		{
			time = null;
		}
		finally
		{
		}
		
		return time;
	}
}

File: GanAndroidApp/app/src/main/java/com/xattacker/gan/service/SystemService.kt
package com.xattacker.gan.service

import com.xattacker.binary.TypeConverter
import com.xattacker.gan.GanAgent
import com.xattacker.gan.data.FunctionType
import com.xattacker.gan.data.ResponsePack
import java.util.*


abstract class SystemService protected constructor(aAgent: GanAgent?) : ServiceFoundation(aAgent)
{
    fun getIPAddress(): String?
    {
        var ip: String? = null
        try
        {
            val response = send(FunctionType.GET_IP, null)
            if (response != null && response.result && response.content != null)
            {
                    ip = String(response.content!!)
            }
        }
        catch (ex: Exception)
        {
            ip = null
        }
        finally
        {
        }
        return ip
    }

    fun getSystemTime(): Date?
    {
        var time: Date? = null
        try
        {
            val response = send(FunctionType.GET_SYSTEM_TIME, null)
            if (response != null && response.result && response.content != null)
            {
                val timestamp: Long = TypeConverter.byteToLong(response.content!!)
                time = Date(timestamp)
            }
        }
        catch (ex: Exception)
        {
            time = null
        }
        finally
        {
        }
        return time
    }
}